Index: eXoApplication/mail/service/src/main/java/org/exoplatform/mail/service/MimeMessageParser.java
===================================================================
--- eXoApplication/mail/service/src/main/java/org/exoplatform/mail/service/MimeMessageParser.java	(revision 75849)
+++ eXoApplication/mail/service/src/main/java/org/exoplatform/mail/service/MimeMessageParser.java	(working copy)
@@ -233,7 +233,7 @@
   }
   
   public static String[] getInvitationHeader(javax.mail.Message message) throws Exception {
-    String[] exoInvitationHeaders = message.getHeader("X-Exo-Invitation");
+    String[] exoInvitationHeaders = message.getHeader("x-exo-invitation".toLowerCase());
     if (exoInvitationHeaders != null) return exoInvitationHeaders ;
     return null ;
   }
Index: eXoApplication/mail/webapp/src/main/java/org/exoplatform/mail/MailUtils.java
===================================================================
--- eXoApplication/mail/webapp/src/main/java/org/exoplatform/mail/MailUtils.java	(revision 75849)
+++ eXoApplication/mail/webapp/src/main/java/org/exoplatform/mail/MailUtils.java	(working copy)
@@ -74,6 +74,7 @@
     final public static String SUPPORTED_VIEW_TYPE_ATTACH[] = {"gif", "png", "jpg", "jpec", "bmp"} ;
     final public static String SIMPLECHARACTER[] = {GREATER_THAN, SMALLER_THAN, "'", "\""};
     final public static int MAX_POPUP_WIDTH = 900;
+    public static final String EXO_INVITATION = "x-exo-invitation".intern();
 
     static public MailService getMailService() throws Exception {
       return (MailService)PortalContainer.getComponent(MailService.class) ;
@@ -270,13 +271,13 @@
     }
 
     public static boolean isInvitation(Message msg) throws Exception {
-      return (msg.getHeader("X-Exo-Invitation") != null) ;
+      return (msg.getHeader(EXO_INVITATION) != null) ;
     }
 
     public static String getEventFrom(Message msg) throws Exception {
       String from = null;
       if (isInvitation(msg)) {
-        from = msg.getHeader("X-Exo-Invitation").split(";")[0].trim() ;
+        from = msg.getHeader(EXO_INVITATION).split(";")[0].trim() ;
       }
       return from ;
     }
@@ -284,7 +285,7 @@
     public static String getEventTo(Message msg) throws Exception {
       String to = null;
       if (isInvitation(msg)) {
-        to = msg.getHeader("X-Exo-Invitation").split(";")[1].trim() ;
+        to = msg.getHeader(EXO_INVITATION).split(";")[1].trim() ;
       }
       return to ;
     }
@@ -300,7 +301,7 @@
     public static String getEventType(Message msg) throws Exception {
       String eventType = null;
       if (isInvitation(msg)) {
-        eventType = msg.getHeader("X-Exo-Invitation").split(";")[2].trim() ;
+        eventType = msg.getHeader(EXO_INVITATION).split(";")[2].trim() ;
       }
       return eventType ;
     }
@@ -308,7 +309,7 @@
     public static String getCalendarId(Message msg) throws Exception {
       String calId = null;
       if (isInvitation(msg)) {
-        calId = msg.getHeader("X-Exo-Invitation").split(";")[3].trim() ;
+        calId = msg.getHeader(EXO_INVITATION).split(";")[3].trim() ;
       }
       return calId ;
     }
@@ -316,7 +317,7 @@
     public static String getCalendarEventId(Message msg) throws Exception {
       String calEvenId = null;
       if (isInvitation(msg)) {
-        calEvenId = msg.getHeader("X-Exo-Invitation").split(";")[4].trim() ;
+        calEvenId = msg.getHeader(EXO_INVITATION).split(";")[4].trim() ;
       }
       return calEvenId ;
     }
Index: eXoApplication/calendar/webapp/src/main/java/org/exoplatform/calendar/CalendarUtils.java
===================================================================
--- eXoApplication/calendar/webapp/src/main/java/org/exoplatform/calendar/CalendarUtils.java	(revision 75849)
+++ eXoApplication/calendar/webapp/src/main/java/org/exoplatform/calendar/CalendarUtils.java	(working copy)
@@ -142,7 +142,7 @@
   public static final String DATETIMEFORMAT = DATEFORMAT +" " +TIMEFORMAT;   
   public static final int DEFAULT_TIMEITERVAL = 15;
   public static final long MILISECONS_OF_DAY = 24*60*59*1000;
-  public static final String EXO_INVITATION = "X-Exo-Invitation".intern();
+  public static final String EXO_INVITATION = "x-exo-invitation".intern();
   public static final String SPECIALCHARACTER[] = {SEMICOLON,COMMA,SLASH,BACKSLASH,
     SINGLE_QUOTE,OR,GREATER_THAN,SMALLER_THAN,QUOTE, QUESTION_MARK, EXCLAMATION, 
     AT, SHARP, MONEY_MARK, PERCENT,EXPONENT,AND,STAR};
@@ -259,22 +259,32 @@
   }
   
   /**
+   * get calendar by user setting (timezone, first day of week)
+   * @param calendarSetting
+   * @return calendar object
+   */
+  public static Calendar getCalendarInstanceBySetting(final CalendarSetting calendarSetting) {
+    Calendar  calendar = GregorianCalendar.getInstance() ;
+    calendar.setLenient(false);
+    calendar.setTimeZone(TimeZone.getTimeZone(calendarSetting.getTimeZone()));
+    calendar.setFirstDayOfWeek(Integer.parseInt(calendarSetting.getWeekStartOn()));
+    return calendar;
+  }
+  
+  /**
    * 
    * @return return an instance of Calendar class which contains user's setting, such as, time zone, first day of week.
    */
-  public static Calendar getInstanceOfCurrentCalendar() { 
-    Calendar  calendar = GregorianCalendar.getInstance() ;
-    calendar.setLenient(false);
+  public static Calendar getInstanceOfCurrentCalendar() {
      try {
       CalendarSetting setting = getCurrentUserCalendarSetting();
-      calendar.setTimeZone(TimeZone.getTimeZone(setting.getTimeZone()));
-      calendar.setFirstDayOfWeek(Integer.parseInt(setting.getWeekStartOn())); 
+      return getCalendarInstanceBySetting(setting); 
     } catch (Exception e) {
-      log.warn(e);
+      if (log.isWarnEnabled()) log.warn("Could not get calendar setting!", e);
+      Calendar calendar = GregorianCalendar.getInstance() ;
+      calendar.setLenient(false);
+      return calendar;
     }
-//    int gmtoffset = calendar.get(Calendar.DST_OFFSET) + calendar.get(Calendar.ZONE_OFFSET);
-//    calendar.setTimeInMillis(System.currentTimeMillis() - gmtoffset) ; 
-    return  calendar;
   }
   public static List<SelectItemOption<String>> getTimesSelectBoxOptions(String timeFormat) {
     WebuiRequestContext context = WebuiRequestContext.getCurrentInstance() ;
Index: eXoApplication/calendar/webapp/src/main/java/org/exoplatform/calendar/webui/UICalendarView.java
===================================================================
--- eXoApplication/calendar/webapp/src/main/java/org/exoplatform/calendar/webui/UICalendarView.java	(revision 75849)
+++ eXoApplication/calendar/webapp/src/main/java/org/exoplatform/calendar/webui/UICalendarView.java	(working copy)
@@ -187,7 +187,7 @@
 //    calendar.setFirstDayOfWeek(Integer.parseInt(calendarSetting_.getWeekStartOn()));
 //    return calendar;
     
-    return  CalendarUtils.getInstanceOfCurrentCalendar();
+    return  CalendarUtils.getCalendarInstanceBySetting(calendarSetting_);
   } 
   public void applySeting() throws Exception {
     displayTimes_ = null ;
@@ -201,6 +201,10 @@
       calendarSetting_ = calService.getCalendarSetting(username) ;
     } 
     dateTimeFormat_ = getDateFormat() + " " + getTimeFormat() ;
+    Date selectedDate = calendar_ != null ? calendar_.getTime() : null;
+    calendar_ = CalendarUtils.getCalendarInstanceBySetting(calendarSetting_);
+    if (selectedDate != null) 
+      calendar_.setTime(selectedDate);
   }
   public void setViewType(String viewType) { this.viewType_ = viewType ; }
   public String getViewType() { return viewType_ ; }
Index: eXoApplication/calendar/webapp/src/main/java/org/exoplatform/calendar/webui/UICalendarContainer.java
===================================================================
--- eXoApplication/calendar/webapp/src/main/java/org/exoplatform/calendar/webui/UICalendarContainer.java	(revision 75849)
+++ eXoApplication/calendar/webapp/src/main/java/org/exoplatform/calendar/webui/UICalendarContainer.java	(working copy)
@@ -17,6 +17,7 @@
 package org.exoplatform.calendar.webui;
 
 import org.exoplatform.webui.config.annotation.ComponentConfig;
+import org.exoplatform.webui.core.UIComponent;
 import org.exoplatform.webui.core.UIContainer;
 import org.exoplatform.webui.core.lifecycle.UIContainerLifecycle;
 
@@ -35,5 +36,12 @@
     addChild(UISearchForm.class, null, null) ;
     addChild(UIMiniCalendar.class, null, null) ;
     addChild(UICalendars.class, null, null) ;    
-  } 
+  }
+  
+  public void applySeting() throws Exception {
+    for(UIComponent comp : getChildren()) {
+      if((comp instanceof CalendarView)) ((CalendarView)comp).applySeting() ;  
+    }
+  }
+  
 }
Index: eXoApplication/calendar/webapp/src/main/java/org/exoplatform/calendar/webui/popup/UICalendarSettingForm.java
===================================================================
--- eXoApplication/calendar/webapp/src/main/java/org/exoplatform/calendar/webui/popup/UICalendarSettingForm.java	(revision 75849)
+++ eXoApplication/calendar/webapp/src/main/java/org/exoplatform/calendar/webui/popup/UICalendarSettingForm.java	(working copy)
@@ -31,6 +31,7 @@
 import org.exoplatform.calendar.service.Utils;
 import org.exoplatform.calendar.service.impl.NewUserListener;
 import org.exoplatform.calendar.webui.UIActionBar;
+import org.exoplatform.calendar.webui.UICalendarContainer;
 import org.exoplatform.calendar.webui.UICalendarPortlet;
 import org.exoplatform.calendar.webui.UICalendarView;
 import org.exoplatform.calendar.webui.UICalendarViewContainer;
@@ -311,6 +312,8 @@
       uiViewContainer.applySeting() ;
       uiViewContainer.refresh() ;
       
+      UICalendarContainer uiCalendarContainer = calendarPortlet.findFirstComponentOfType(UICalendarContainer.class);
+      uiCalendarContainer.applySeting();
       // TODO CS-4165
       calendarPortlet.findFirstComponentOfType(UICalendarView.class).setCalendarSetting(calendarSetting);
       
Index: eXoApplication/calendar/webapp/src/main/webapp/templates/calendar/webui/UIWeekView.gtmpl
===================================================================
--- eXoApplication/calendar/webapp/src/main/webapp/templates/calendar/webui/UIWeekView.gtmpl	(revision 75849)
+++ eXoApplication/calendar/webapp/src/main/webapp/templates/calendar/webui/UIWeekView.gtmpl	(working copy)
@@ -7,6 +7,7 @@
 	import org.exoplatform.calendar.CalendarUtils;
 	import java.util.Locale;
 	import org.exoplatform.webui.application.WebuiRequestContext;
+	import java.text.ParseException;
 	uiform.begin()
  	_ctx.include("app:/templates/calendar/webui/UIHeaderBar.gtmpl");
  	Calendar currentCalendar = uicomponent.getCurrentCalendar() ;
@@ -248,7 +249,14 @@
 								//df.setCalendar(cl) ;
 					 			startTime = dayFormat.format(cl.getTime()) + " " + time;
 					 			//dtf.setCalendar(cl) ;
-					 			cl.setTime(fullDateFormat.parse(startTime)) ;
+					 			fullDateFormat.setLenient(false);
+					 			// add try-catch block to handle Daylight Saving Time problem 
+					 			try {
+					 			  cl.setTime(fullDateFormat.parse(startTime)) ;
+					 			} catch (ParseException e) {
+					 			  fullDateFormat.setLenient(true);
+					 			  cl.setTime(fullDateFormat.parse(startTime));   
+					 			}
 					 			dayOfWeek = cl.get(Calendar.DAY_OF_WEEK) ;
 					 		  if(uicomponent.isCurrentDay(cl.get(Calendar.DATE), cl.get(Calendar.MONTH), cl.get(Calendar.YEAR))) {
 						  		cssClass = "Today" ;
