Index: eXoApplication/mail/service/src/main/java/org/exoplatform/mail/service/MimeMessageParser.java
===================================================================
--- eXoApplication/mail/service/src/main/java/org/exoplatform/mail/service/MimeMessageParser.java	(revision 75849)
+++ eXoApplication/mail/service/src/main/java/org/exoplatform/mail/service/MimeMessageParser.java	(working copy)
@@ -233,7 +233,7 @@
   }
   
   public static String[] getInvitationHeader(javax.mail.Message message) throws Exception {
-    String[] exoInvitationHeaders = message.getHeader("X-Exo-Invitation");
+    String[] exoInvitationHeaders = message.getHeader("x-exo-invitation".toLowerCase());
     if (exoInvitationHeaders != null) return exoInvitationHeaders ;
     return null ;
   }
Index: eXoApplication/mail/webapp/src/main/java/org/exoplatform/mail/MailUtils.java
===================================================================
--- eXoApplication/mail/webapp/src/main/java/org/exoplatform/mail/MailUtils.java	(revision 75849)
+++ eXoApplication/mail/webapp/src/main/java/org/exoplatform/mail/MailUtils.java	(working copy)
@@ -74,6 +74,7 @@
     final public static String SUPPORTED_VIEW_TYPE_ATTACH[] = {"gif", "png", "jpg", "jpec", "bmp"} ;
     final public static String SIMPLECHARACTER[] = {GREATER_THAN, SMALLER_THAN, "'", "\""};
     final public static int MAX_POPUP_WIDTH = 900;
+    public static final String EXO_INVITATION = "x-exo-invitation".intern();
 
     static public MailService getMailService() throws Exception {
       return (MailService)PortalContainer.getComponent(MailService.class) ;
@@ -270,13 +271,13 @@
     }
 
     public static boolean isInvitation(Message msg) throws Exception {
-      return (msg.getHeader("X-Exo-Invitation") != null) ;
+      return (msg.getHeader(EXO_INVITATION) != null) ;
     }
 
     public static String getEventFrom(Message msg) throws Exception {
       String from = null;
       if (isInvitation(msg)) {
-        from = msg.getHeader("X-Exo-Invitation").split(";")[0].trim() ;
+        from = msg.getHeader(EXO_INVITATION).split(";")[0].trim() ;
       }
       return from ;
     }
@@ -284,7 +285,7 @@
     public static String getEventTo(Message msg) throws Exception {
       String to = null;
       if (isInvitation(msg)) {
-        to = msg.getHeader("X-Exo-Invitation").split(";")[1].trim() ;
+        to = msg.getHeader(EXO_INVITATION).split(";")[1].trim() ;
       }
       return to ;
     }
@@ -300,7 +301,7 @@
     public static String getEventType(Message msg) throws Exception {
       String eventType = null;
       if (isInvitation(msg)) {
-        eventType = msg.getHeader("X-Exo-Invitation").split(";")[2].trim() ;
+        eventType = msg.getHeader(EXO_INVITATION).split(";")[2].trim() ;
       }
       return eventType ;
     }
@@ -308,7 +309,7 @@
     public static String getCalendarId(Message msg) throws Exception {
       String calId = null;
       if (isInvitation(msg)) {
-        calId = msg.getHeader("X-Exo-Invitation").split(";")[3].trim() ;
+        calId = msg.getHeader(EXO_INVITATION).split(";")[3].trim() ;
       }
       return calId ;
     }
@@ -316,7 +317,7 @@
     public static String getCalendarEventId(Message msg) throws Exception {
       String calEvenId = null;
       if (isInvitation(msg)) {
-        calEvenId = msg.getHeader("X-Exo-Invitation").split(";")[4].trim() ;
+        calEvenId = msg.getHeader(EXO_INVITATION).split(";")[4].trim() ;
       }
       return calEvenId ;
     }
Index: eXoApplication/calendar/webapp/src/main/java/org/exoplatform/calendar/CalendarUtils.java
===================================================================
--- eXoApplication/calendar/webapp/src/main/java/org/exoplatform/calendar/CalendarUtils.java	(revision 75849)
+++ eXoApplication/calendar/webapp/src/main/java/org/exoplatform/calendar/CalendarUtils.java	(working copy)
@@ -142,7 +142,7 @@
   public static final String DATETIMEFORMAT = DATEFORMAT +" " +TIMEFORMAT;   
   public static final int DEFAULT_TIMEITERVAL = 15;
   public static final long MILISECONS_OF_DAY = 24*60*59*1000;
-  public static final String EXO_INVITATION = "X-Exo-Invitation".intern();
+  public static final String EXO_INVITATION = "x-exo-invitation".intern();
   public static final String SPECIALCHARACTER[] = {SEMICOLON,COMMA,SLASH,BACKSLASH,
     SINGLE_QUOTE,OR,GREATER_THAN,SMALLER_THAN,QUOTE, QUESTION_MARK, EXCLAMATION, 
     AT, SHARP, MONEY_MARK, PERCENT,EXPONENT,AND,STAR};
