Index: packaging/module/src/main/javascript/cs.packaging.module.js
===================================================================
--- packaging/module/src/main/javascript/cs.packaging.module.js	(revision 79111)
+++ packaging/module/src/main/javascript/cs.packaging.module.js	(working copy)
@@ -61,9 +61,6 @@
     new Project("org.exoplatform.cs", "exo.cs.eXoApplication.chat.webapp", "war", module.version).
     addDependency(new Project("org.exoplatform.cs", "exo.cs.eXoApplication.chat.service", "jar", module.version).
     addDependency(new Project("org.exoplatform.cs", "exo.cs.eXoApplication.organization.service", "jar", module.version)).
-    //addDependency(new Project("org.exoplatform.cs", "exo.cs.eXoApplication.organization.webapp", "war", module.version)).
-    addDependency(new Project("org.exoplatform.cs", "exo.cs.eXoApplication.organization.client.openfire", "jar", module.version)).
-    //addDependency(new Project("org.exoplatform.cs", "exo.cs.eXoApplication.organization.webapp", "war", module.version)).
     addDependency(new Project("jivesoftware", "smack", "jar", "${jivesoftware.smack.version}")).
     addDependency(new Project("jivesoftware", "smackx", "jar", "${jivesoftware.smackx.version}")).
     addDependency(new Project("org.jcrom", "jcrom", "jar", "${jcrom.version}")).
@@ -105,6 +102,10 @@
 		
   module.server.jboss.patchear = 
 	    new Project("org.exoplatform.cs", "exo.cs.server.jboss.patch-ear", "jar", module.version);
+  
+  module.server.openfire = {};
+  module.server.openfire.patch = 
+    new Project("org.exoplatform.cs", "exo.cs.eXoApplication.organization.client.openfire", "jar", module.version);
       
   // CS demo 
    module.demo = {};
@@ -147,16 +148,7 @@
 
   // We use only the local repository which must be defined in the repos list
   server.openfireJarPath = new java.net.URL(eXo.env.m2Repos[0]).getPath();
-  
-  for (var i=0; i < module.eXoApplication.chat.dependencies.size(); i++) {
-	  var tmpObj = module.eXoApplication.chat.dependencies.get(i).dependencies; 
-	  for (var j=0; j<tmpObj.size(); j++) {
-		  if (tmpObj.get(j).artifactId == "exo.cs.eXoApplication.organization.client.openfire") {
-			  server.openfireJarPath += "/" + tmpObj.get(j).relativePath;
-			  break;
-		  }
-	  }
-  }
+  server.openfireJarPath += "/" + module.server.openfire.patch.relativePath;
    
   server.cleanServer = "openfire-" + "${openfire.version}";
   server.name = "exo-chatserver";
