Index: eXoApplication/calendar/webapp/src/main/webapp/templates/calendar/webui/UIMonthView.gtmpl
===================================================================
--- eXoApplication/calendar/webapp/src/main/webapp/templates/calendar/webui/UIMonthView.gtmpl	(revision 72608)
+++ eXoApplication/calendar/webapp/src/main/webapp/templates/calendar/webui/UIMonthView.gtmpl	(working copy)
@@ -42,7 +42,6 @@
 	tf.setCalendar(CalendarUtils.getInstanceTempCalendar()) ;
 	DateFormat dtf = new SimpleDateFormat(dateTimeFormat, locale) ;
 	dtf.setCalendar(CalendarUtils.getInstanceTempCalendar()) ;
-	DateFormat wf = new  SimpleDateFormat("E", locale) ;
 	DateFormat tempFormat = new  SimpleDateFormat("EEE MMM dd yyyy HH:mm:ss", Locale.ENGLISH) ;
 	tempFormat.setCalendar(uicomponent.getInstanceTempCalendar()) ;
 	%>
@@ -79,16 +78,16 @@
 
 					<%
 			  dfs_ = new DateFormatSymbols(locale) ;
-			  calendar = uicomponent.getInstanceTempCalendar();
-	      dayStart = uicomponent.getCalendarSetting().getWeekStartOn();
-	      calendar.set(Calendar.DAY_OF_WEEK, Integer.parseInt(dayStart));
-	      for(int i = 1; i < dfs_.getWeekdays().length ; i ++) {
+			  def aCalendar = uicomponent.getInstanceTempCalendar();
+        String[] dayNames = dfs_.getShortWeekdays();
+        aCalendar.set(Calendar.DAY_OF_WEEK, aCalendar.getFirstDayOfWeek());
+	      for(int i = 0; i < 7; i ++) {
 					String cssStyle = "" ;
-					if(calendar.get(Calendar.DAY_OF_WEEK) == Calendar.SUNDAY) cssStyle = "color: red;" ;
+					if(aCalendar.get(Calendar.DAY_OF_WEEK) == Calendar.SUNDAY) cssStyle = "color: red;" ;
 				%>
-			<div class="WeekTitle" style="$cssStyle"><%=wf.format(calendar.getTime())%></div>
+			<div class="WeekTitle" style="$cssStyle"><%=dayNames[aCalendar.get(Calendar.DAY_OF_WEEK)]%></div>
 			  <%
-			  	calendar.add(Calendar.DAY_OF_WEEK, 1);
+			  	aCalendar.add(Calendar.DAY_OF_WEEK, 1);
 	      }
 	      %>			
   				<div style="clear: left;"><span></span></div>
Index: eXoApplication/calendar/webapp/src/main/webapp/templates/calendar/webui/UIMiniCalendar.gtmpl
===================================================================
--- eXoApplication/calendar/webapp/src/main/webapp/templates/calendar/webui/UIMiniCalendar.gtmpl	(revision 72608)
+++ eXoApplication/calendar/webapp/src/main/webapp/templates/calendar/webui/UIMiniCalendar.gtmpl	(working copy)
@@ -29,8 +29,6 @@
 	DateFormat sdtf = new  SimpleDateFormat("MM/d", locale) ;
  	sdtf.setCalendar(uicomponent.getInstanceTempCalendar()) ;
 	Map data = uicomponent.getData() ;
-	
-  DateFormat wf = new  SimpleDateFormat("E", locale) ;
 %>
 <div class="UIMiniCalendar">
 	<div class="MiniCalendarContainer">
@@ -80,13 +78,13 @@
 			  <%
 			  dfs_ = new DateFormatSymbols(locale) ;
 			  Calendar calendar = uicomponent.getInstanceTempCalendar();
-	      String dayStart = uicomponent.getCalendarSetting().getWeekStartOn();
-	      calendar.set(Calendar.DAY_OF_WEEK, Integer.parseInt(dayStart));
-	      for(int i = 1; i < dfs_.getWeekdays().length ; i ++) {
+        String[] dayNames = dfs_.getShortWeekdays();
+        calendar.set(Calendar.DAY_OF_WEEK, calendar.getFirstDayOfWeek());
+	      for(int i = 0; i < 7; i++) {
 			  	cssStyle = "DayTitle" ;
 					if(calendar.get(Calendar.DAY_OF_WEEK) == Calendar.SUNDAY) cssStyle = "SundayTitle" ;
 				%>
-			<div class="$cssStyle"><%=wf.format(calendar.getTime()).substring(0,1)%></div>
+			<div class="$cssStyle"><%=dayNames[calendar.get(Calendar.DAY_OF_WEEK)].substring(0,1)%></div>
 			  <%
 			  	calendar.add(Calendar.DAY_OF_WEEK, 1);
 	      }
Index: eXoApplication/calendar/webapp/src/main/webapp/templates/calendar/webui/UIYearView.gtmpl
===================================================================
--- eXoApplication/calendar/webapp/src/main/webapp/templates/calendar/webui/UIYearView.gtmpl	(revision 72608)
+++ eXoApplication/calendar/webapp/src/main/webapp/templates/calendar/webui/UIYearView.gtmpl	(working copy)
@@ -78,13 +78,13 @@
 									<%
 										dfs_ = new DateFormatSymbols(locale) ;
 										Calendar calendar = uicomponent.getInstanceTempCalendar();
-										String dayStart = uicomponent.getCalendarSetting().getWeekStartOn();
-										calendar.set(Calendar.DAY_OF_WEEK, Integer.parseInt(dayStart));
-										for(int i = 1; i < dfs_.getWeekdays().length ; i ++) {
+										String[] dayNames = dfs_.getShortWeekdays();
+										calendar.set(Calendar.DAY_OF_WEEK, calendar.getFirstDayOfWeek());
+										for(int i = 0; i < 7; i++) {
 											cssClass = "DayTitle" ;
 											if(calendar.get(Calendar.DAY_OF_WEEK) == Calendar.SUNDAY) cssClass = "SundayTitle" ;
 										%>
-									<div class="$cssClass"><%=wf.format(calendar.getTime()).substring(0,1)%></div>
+									<div class="$cssClass"><%=dayNames[calendar.get(Calendar.DAY_OF_WEEK)].substring(0,1)%></div>
 										<%
 											calendar.add(Calendar.DAY_OF_WEEK, 1);
 										}
