Index: eXoApplication/calendar/webapp/src/main/java/org/exoplatform/calendar/webui/popup/UIQuickAddEvent.java
===================================================================
--- eXoApplication/calendar/webapp/src/main/java/org/exoplatform/calendar/webui/popup/UIQuickAddEvent.java	(revision 73687)
+++ eXoApplication/calendar/webapp/src/main/java/org/exoplatform/calendar/webui/popup/UIQuickAddEvent.java	(working copy)
@@ -164,7 +164,6 @@
       return df.parse(fromField.getValue() + " " + timeFile.getValue()) ;
     }
     catch (Exception e) {
-      e.printStackTrace() ;
       return null ;
     }
   }
@@ -201,7 +200,6 @@
       return df.parse(toField.getValue() + " " + timeFile.getValue()) ;
       
     } catch (Exception e) {
-      e.printStackTrace() ;
       return null ;
     }
   }
