Index: eXoApplication/calendar/webapp/src/main/webapp/javascript/eXo/calendar/UIWeekView.js
===================================================================
--- eXoApplication/calendar/webapp/src/main/webapp/javascript/eXo/calendar/UIWeekView.js	(revision 70920)
+++ eXoApplication/calendar/webapp/src/main/webapp/javascript/eXo/calendar/UIWeekView.js	(working copy)
@@ -83,6 +83,11 @@
 	var UICalendarPortlet = eXo.calendar.UICalendarPortlet ;
 	var inter = UICalendarPortlet.getInterval(el) ;
 	if (el.length <= 0) return ;
+    // set position of events in vertical axis.
+    for (var i = 0; i < el.length; i++) {
+    	UICalendarPortlet.setSize(el[i]);
+    }
+	
 	var width = "" ;
 	for(var i = 0 ; i < inter.length ; i ++) {
 		var totalWidth = (arguments.length > 1) ? arguments[1] : parseFloat(100) ;
@@ -108,7 +113,6 @@
 		}
 		var n = 0 ;
 		for(var j = inter[i]; j < inter[i+1] ; j++) {
-			eXo.calendar.UICalendarPortlet.setSize(el[j]) ;
 			if(mark != null) {				
 				width = parseFloat((totalWidth + left - parseFloat(el[mark].style.left) - parseFloat(el[mark].style.width))/len) ;
 			} else {
