Index: eXoApplication/mail/service/src/main/java/org/exoplatform/mail/service/MailSetting.java
===================================================================
--- eXoApplication/mail/service/src/main/java/org/exoplatform/mail/service/MailSetting.java	(revision 80131)
+++ eXoApplication/mail/service/src/main/java/org/exoplatform/mail/service/MailSetting.java	(working copy)
@@ -77,7 +77,7 @@
 
   private long                periodCheckAuto_    = 0;
 
-  private String              defaultAccount_     = StringUtils.EMPTY;
+  private String              defaultAccount_     ;
 
   private boolean             useWysiwyg_         = true;
 
