Index: eXoApplication/calendar/service/src/main/java/org/exoplatform/calendar/service/impl/JCRDataStorage.java
===================================================================
--- eXoApplication/calendar/service/src/main/java/org/exoplatform/calendar/service/impl/JCRDataStorage.java	(revision 79171)
+++ eXoApplication/calendar/service/src/main/java/org/exoplatform/calendar/service/impl/JCRDataStorage.java	(working copy)
@@ -2713,6 +2713,8 @@
   public Map<String,String> checkFreeBusy(EventQuery eventQuery, Date date) throws Exception {
     Node eventFolder = getEventFolder(date);
     Map<String, String> participantMap = new HashMap<String, String>() ;
+    String calendarPath = eventQuery.getCalendarPath();
+    String[] orderBy = eventQuery.getOrderBy();
     eventQuery.setCalendarPath(eventFolder.getPath()) ;
     eventQuery.setOrderBy(new String[]{Utils.EXO_FROM_DATE_TIME}) ;
     QueryManager qm =  getSession(createSystemProvider()).getWorkspace().getQueryManager();
@@ -2722,6 +2724,7 @@
     String from ;
     String to ;
     for(String par : pars) {
+      String[] participants = eventQuery.getParticipants();
       eventQuery.setParticipants(new String[]{par}) ;
       query = qm.createQuery(eventQuery.getQueryStatement(), Query.XPATH);
       QueryResult result = query.execute();
@@ -2748,7 +2751,10 @@
           participantMap.put(par, timeValues.toString()) ;
         }
       }
+      eventQuery.setParticipants(participants);
     }
+    eventQuery.setCalendarPath(calendarPath);
+    eventQuery.setOrderBy(orderBy);
     return participantMap ;
   }
 
