Index: eXoApplication/chatbar/webapp/src/main/webapp/javascript/eXo/communication/chat/webui/UIChatWindow.js
===================================================================
--- eXoApplication/chatbar/webapp/src/main/webapp/javascript/eXo/communication/chat/webui/UIChatWindow.js	(revision 48774)
+++ eXoApplication/chatbar/webapp/src/main/webapp/javascript/eXo/communication/chat/webui/UIChatWindow.js	(working copy)
@@ -923,6 +923,7 @@
   var thys = eXo.communication.chatbar.webui.UIChatWindow;
   thys._isOnLoading = true;
   var visible = thys._isVisible();
+  thys.rootNode.style.visibility = "hidden";
   thys.initSession();
   var tabList = thys._getOption('tabs');
   if (tabList) {
@@ -955,6 +956,7 @@
   }
   thys.setVisible(visible, null, true);
   thys._isOnLoading = false;
+  thys.rootNode.style.visibility = "visible";
 };
 
 /**
