Index: eXoApplication/calendar/webapp/src/main/webapp/templates/calendar/webui/UIListView.gtmpl
===================================================================
--- eXoApplication/calendar/webapp/src/main/webapp/templates/calendar/webui/UIListView.gtmpl	(revision 80153)
+++ eXoApplication/calendar/webapp/src/main/webapp/templates/calendar/webui/UIListView.gtmpl	(working copy)
@@ -20,7 +20,16 @@
    Locale locale = context.getParentAppRequestContext().getLocale() ;
   dateTimeFormat = uicomponent.getDateTimeFormat() ;
   DateFormat df = new SimpleDateFormat(dateTimeFormat, locale) ;
-  _ctx.include("app:/templates/calendar/webui/UIHeaderBar.gtmpl");
+  boolean searchDisplay = uicomponent.isDisplaySearchResult();
+  if(searchDisplay) {
+%>
+    <div class="Result ListBar">
+      <%=uiform.getLabel('searchResult')%>
+    </div>
+<%
+  }else {
+    _ctx.include("app:/templates/calendar/webui/UIHeaderBar.gtmpl");
+  }
   selectedEvent = uicomponent.getSelectedEvent() ;
   def rcontext = _ctx.getRequestContext() ;
   rcontext.getJavascriptManager().addJavascript('eXo.calendar.UICalendarPortlet.listViewDblClick("'+uiform.id+'") ;') ;
@@ -34,7 +43,10 @@
     <table cellspacing="0" borderspacing="0" id="UIListUsers" class="UIGrid">
       <thead>
         <tr>
-          <th style="width: 30px; padding-left: 4px;"><input style="float: left; margin-top: 9px; margin-left: 6px;" type="checkbox" class="checkbox" title="<%=uiform.getLabel('CheckAll')%>" value="4"/></th>
+        <%if(!searchDisplay) {
+        %>
+           <th style="width: 30px; padding-left: 4px;"><input style="float: left; margin-top: 9px; margin-left: 6px;" type="checkbox" class="checkbox" title="<%=uiform.getLabel('CheckAll')%>" value="4"/></th>
+        <%}%>
           <th style="width: 24px;"><div class="AttachmentIcon">&nbsp;</div></th>
           <%
             String sortLink = uicomponent.event("Sort", uicomponent.EVENT_PRIORITY.toString()) ;   
@@ -94,7 +106,9 @@
 			  def color = uicomponent.getColors().get(event.getCalType() + CalendarUtils.COLON + event.getCalendarId()) ;
           %>
               <tr actionLink="$actionLink" onclick="eXo.calendar.UICalendarPortlet.listViewClickCallback(this);" class="UIListViewRow" style="$rowStyle" eventid="<%=event.getId()%>" calid="<%=event.getCalendarId()%>" calType=<%=event.getCalType()%>>
+              <%if(!searchDisplay) {%>
                 <td><%uicomponent.renderField(uicomponent.getChildById(event.getId()))%></td>
+              <%}%>
                 <td style="padding: 0px 0px 0px 7px;"><div class="$attachIcon" >&nbsp;</div></td>
                 <td class="text"><div class="<%=event.getPriority()%>PriorityIcon">&nbsp;</div></td>      
                 <td class="text">
@@ -125,9 +139,7 @@
       }
     %>
   </div>
-  <%
-    if(uicomponent.isDisplaySearchResult()) {
-  %>
+  <%if(searchDisplay) {%>
     <div class="UIAction">
       <a href="<%= uicomponent.event('CloseSearch')%>" class="ActionButton LightBlueStyle"><%=uiform.getLabel('CloseSearch')%></a>
     </div>
